x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceMatcher.java

```diff
@@ -285,6 +290,11 @@ public void clear() {
         completed.clear();
     }
 
+    @Override
+    public long ramBytesUsed() {
+        return SHALLOW_SIZE + RamUsageEstimator.sizeOf(keyToSequences) + RamUsageEstimator.sizeOf(stageToKeys);
```

Comment:
One needs to add `completed` which represents the completed sequences.
While the other two structures represent the in-flight sequences.
I'm not sure whether we want to differentiate between them or not (and label them appropriately since the results are typically significantly smaller than the intermediate data).